linera-views/src/backends/dual.rs

```diff
@@ -226,59 +220,79 @@ where
     }
 }
 
-impl<S1, S2, A> AdminKeyValueStore for DualStore<S1, S2, A>
+impl<D1, D2, A> KeyValueDatabase for DualDatabase<D1, D2, A>
 where
-    S1: AdminKeyValueStore,
-    S2: AdminKeyValueStore,
+    D1: KeyValueDatabase,
+    D2: KeyValueDatabase,
     A: DualStoreRootKeyAssignment,
 {
-    type Config = DualStoreConfig<S1::Config, S2::Config>;
+    type Config = DualStoreConfig<D1::Config, D2::Config>;
+    type Store = DualStore<D1::Store, D2::Store>;
 
     fn get_name() -> String {
-        format!("dual {} and {}", S1::get_name(), S2::get_name())
+        format!("dual {} and {}", D1::get_name(), D2::get_name())
     }
 
     async fn connect(config: &Self::Config, namespace: &str) -> Result<Self, Self::Error> {
-        let first_store = S1::connect(&config.first_config, namespace)
+        let first_database = D1::connect(&config.first_config, namespace)
             .await
             .map_err(DualStoreError::First)?;
-        let second_store = S2::connect(&config.second_config, namespace)
+        let second_database = D2::connect(&config.second_config, namespace)
             .await
             .map_err(DualStoreError::Second)?;
-        let store_in_use = A::assigned_store(&[])?;
-        Ok(Self {
-            first_store,
-            second_store,
-            store_in_use,
+        let database = Self {
+            first_database,
+            second_database,
             _marker: std::marker::PhantomData,
-        })
+        };
+        Ok(database)
     }
 
-    fn open_exclusive(&self, root_key: &[u8]) -> Result<Self, Self::Error> {
-        let first_store = self
-            .first_store
-            .open_exclusive(root_key)
-            .map_err(DualStoreError::First)?;
-        let second_store = self
-            .second_store
-            .open_exclusive(root_key)
-            .map_err(DualStoreError::Second)?;
-        let store_in_use = A::assigned_store(root_key)?;
-        Ok(Self {
-            first_store,
-            second_store,
-            store_in_use,
-            _marker: std::marker::PhantomData,
-        })
+    fn open_exclusive(&self, root_key: &[u8]) -> Result<Self::Store, Self::Error> {
+        match A::assigned_store(root_key)? {
+            StoreInUse::First => {
+                let store = self
+                    .first_database
+                    .open_exclusive(root_key)
+                    .map_err(DualStoreError::First)?;
+                Ok(DualStore::First(store))
+            }
+            StoreInUse::Second => {
+                let store = self
+                    .second_database
+                    .open_exclusive(root_key)
+                    .map_err(DualStoreError::Second)?;
+                Ok(DualStore::Second(store))
+            }
+        }
+    }
```

Comment:
So, now the dual store notion has become a notion orthogonal to the notion of shared/exclusive.
This is a generalization, but my question is why?
Usually, we want to introduce a new notion for some reason.